src/Auth/CanResolvedDirective.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types=1);
+
+namespace Nuwave\Lighthouse\Auth;
+
+use GraphQL\Language\AST\FieldDefinitionNode;
+use GraphQL\Language\AST\InterfaceTypeDefinitionNode;
+use GraphQL\Language\AST\ObjectTypeDefinitionNode;
+use Illuminate\Contracts\Pagination\Paginator;
+use Illuminate\Database\Eloquent\Model;
+use Nuwave\Lighthouse\Exceptions\DefinitionException;
+use Nuwave\Lighthouse\Execution\Resolved;
+use Nuwave\Lighthouse\Execution\ResolveInfo;
+use Nuwave\Lighthouse\Schema\AST\DocumentAST;
+use Nuwave\Lighthouse\Schema\RootType;
+use Nuwave\Lighthouse\Support\Contracts\FieldManipulator;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+use Nuwave\Lighthouse\Support\Utils;
+
+class CanResolvedDirective extends BaseCanDirective implements FieldManipulator
+{
+    public static function definition(): string
+    {
+        $commonArguments = BaseCanDirective::commonArguments();
+
+        return /** @lang GraphQL */ <<<GRAPHQL
+"""
+Check a Laravel Policy to ensure the current user is authorized to access a field.
+
+Check the policy against the model instances returned by the field resolver.
+Only use this if the field does not mutate data, it is run before checking.
+"""
+directive @canResolved(
+{$commonArguments}
+) repeatable on FIELD_DEFINITION
+GRAPHQL;
+    }
+
+    protected function authorizeRequest(mixed $root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo, callable $resolver, callable $authorize): mixed
+    {
+        return Resolved::handle(
+            $resolver($root, $args, $context, $resolveInfo),
+            function ($modelLike) use ($authorize) {
```

Comment:
```suggestion
            function (Paginator|iterable|Model $modelLike) use ($authorize): Paginator|iterable|Model {
```

Reply:
`$modelLike` might be object or array (hash) as well.

Reply:
`array` is a kind of `iterable`, and I don't think it can be any other `object` - given `Utils::applyEach()` requires an `iterable<Model>` or `Model`.

Reply:
Well, the current code won't allow anything but Eloquent Model or list of it. But `authorize` accepts any type of object or array. I think it's better to change `?Model $model` to just `mixed $model` so at least Doctrine plain objects can be used there. Hashes won't work, but fixing it might be tricky, so we might leave it for the future.

Reply:
Alright, I am fine with accepting `mixed` here. A test with some non-`Model` object would be nice.